### PR TITLE
feat: workerId

### PR DESCRIPTION
**Pull request by Aslemammad**
It can be imported from `tinypool` as `workerId` in the worker when running!

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -807,7 +836,7 @@ class ThreadPool {
           taskInfo.workerInfo.taskInfos.delete(taskInfo.taskId)
           if (!taskInfo.workerInfo.taskInfos.size) {
             this._removeWorker(taskInfo.workerInfo)
-            this._ensureMaximumWorkers()
+            // this._ensureMaximumWorkers()
```
Comment (sheremet-va) [inline src/index.ts:839]: are you sure about this? 👀

File: src/index.ts [hunk 2]
```diff
@@ -1036,6 +1065,8 @@ class Tinypool extends EventEmitterAsyncResource {
   }
 }
 
+const _workerId = process.__tinypool_state__?.workerId
+
 export * from './common'
-export { Tinypool, Options }
+export { Tinypool, Options, _workerId as workerId }
```
Comment (sheremet-va) [inline src/index.ts:1071]: So, the only way I can get worker id is by importing `tinypool` inside a script that is running in a worker? Like this:

```ts
import { workerId } from 'tinypool'
```

**Review by sheremet-va — CHANGES_REQUESTED** [review]
Also lockfile is kind of big for this changes, don't you think? 👀

**Comment by sheremet-va** [comment]
Also some tests are failing

File: src/index.ts (hunk 1 above)
Comment (Aslemammad) [reply src/index.ts:839]: Yes, we should not spawn workers, we should let the scheduler figures it out!

Comment (Aslemammad) [reply src/index.ts:839]: Yes, we should not spawn workers, we should let the scheduler figures it out!

File: src/index.ts (hunk 2 above)
Comment (Aslemammad) [reply src/index.ts:1071]: or accessing `process.__tinypool_state__?.workerId`

**Pull request merged**